app/battle/knight.py

```diff
@@ -0,0 +1,44 @@
+class Knight:
+    def __init__(self,
+                 name: str,
+                 power: int,
+                 hp: int,
+                 armour: list,
+                 weapon: dict,
+                 potion: dict,
+                 ) -> None:
+
+        self.name = name
+        self.power = power
+        self.hp = hp
+        self.protection = 0  # Default protection

```

Comment:
Remove this comment, there is no need for it since it is clear that 0 is a default value